packages/fetchai/connections/http_client/connection.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP client connection and channel"""
+
+import asyncio
+import json
+import logging
+from asyncio import CancelledError
+from typing import Optional, Set, Union, cast
+
+import requests
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+SUCCESS = 200
+NOT_FOUND = 404
+REQUEST_TIMEOUT = 408
+SERVER_ERROR = 500
+
+logger = logging.getLogger(__name__)
+
+RequestId = str
+
+
+class HTTPClientChannel:
+    """A wrapper for a HTTPClient for a RESTful API."""
+
+    def __init__(
+        self,
+        agent_address: Address,
+        address: str,
+        port: int,
+        connection_id: PublicId,
+        excluded_protocols: Optional[Set[PublicId]] = None,
+        restricted_to_protocols: Optional[Set[PublicId]] = None,
+    ):
+        """
+        Initialize an http client channel.
+
+        :param agent_address: the address of the agent.
+        :param address: RESTful API hostname / IP address
```

Comment:
to keep the connection generic, we shouldn't refer to REST, sicne it is a particular paradigm built over HTTP.

Reply:
An alternative: HTTP server address?